ext/drbbdup/drbbdup.h

```diff
@@ -145,10 +149,21 @@ typedef void (*drbbdup_destroy_orig_analysis_t)(void *drcontext, void *user_data
  * The user can use thread allocation for storing the analysis result.
  *
  * The analysis data is destroyed via a #drbbdup_analyze_case_t function.
+ *
+ * Unlike for original basic block analysis, this call-back does not return a single
+ * and isolated basic block copy. Instead, it provides the main instruction list that
+ * contains all basic block duplications and is used internally by drbbdup. This enables
+ * users to directly insert label instructions during their analysis.
+ * In order to iterate only over the instructions pertaining to the case, the call-back
+ * provides \p instr which should be used in conjuction with
+ * drbbdup_get_next_analysis_instr(). Analysis labels pertaining to \p instr should
+ * be inserted prior to \p where. This enables, for different cases, the insertion
+ * of analysis labels for instructions such as syscalls and jumps, which cannot
+ * be duplicated in accordance to DR rules.
```

Comment:
As mentioned in an earlier discussion, the drcachesim use cases inserts labels during analysis for optimizing address traces by eliding addresses in the function offline_instru_t::identify_elidable_addresses().  The same analysis code is shared with the post-processor, a standalone app using DR as a library, to reconstruct the address trace and insert the missing addresses.  This makes using a custom iterator not easy: now the code can't be directly shared.

Possible solutions:

1) Use this custom iterator interface, requiring drcachesim to make two versions of its analysis code (can share the loop body; two versions of the outer code, with each calling environment calling its own version).

2) Have drbbdup temporarily NULL out the ends of the case instrlist sequence, and temporarily clone any syscall/cti at the end, allowing normal iteration.  After calling the case analysis cb, drbbdup would then re-stitch the instrlist links, and remove the clone.  I didn't check but I assume there's a meta jump or sthg to go to the shared syscall instr.  Any labels inserted for the syscall would still be in the right place then it would seem.